src/obr/core/core.py

```diff
@@ -118,17 +128,31 @@ def logged_func(func, doc, **kwargs):
         logging.error("Failure" + __file__ + __name__ + func.__name__ + kwargs + str(e))
         state = "failure"
 
-    res = d.get(cmd_str, [])
-    res.append(
-        {
-            "args": str(kwargs),
-            "state": state,
-            "type": "obr",
-            "timestamp": str(datetime.now()),
-        }
-    )
-    d[cmd_str] = res
-    doc["obr"] = d
+    res = {
+        "cmd": cmd_str,
+        "args": str(kwargs),
+        "state": state,
+        "type": "obr",
+        "timestamp": str(datetime.now()),
+        "type": "logged_func",
+        "user": os.environ.get("USER"),
+        "hostname": os.environ.get("HOST"),
+    }
+    doc["history"].append(res)
+
+
+def get_latest_log(job):
+    """find latest"""
+    from ..OpenFOAM.case import OpenFOAMCase
```

Comment:
why not import it at the top level of the file?

Reply:
To avoid circular imports, since case imports core.